onc-healthit/lantern-back-end
# Lantern Endpoint Addition Source List

Changes in this PR:
This PR adds the ability to add one-off FHIR Endpoints to a Lantern endpoint list. If somebody reports an endpoint directly to Lantern that is not from a specific list, it can now be added to the LanternEndpointSources.json file in the resources/dev_resources and resources/prod_resources directories, which as of right now are empty. The format for these lists has been added to tne endpoint manager README. This PR also adds a new parser for the Lantern endpoint list, and updates some of the existing tests to make sure the Lantern endpoint source parser works correctly.

TO TEST:
Copy the contents of endpointmanger/resources/LanternEndpointSources.json into resources/dev_resources/LanternEndpointSources.json. When you run make populatedb, you should see the fake endpoints in that file in the fhir_endpoints table with a list source of LanternEndpointSourcesJson

Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

**Submitter:**
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: [Lantern 251 JIRA Ticket](https://oncprojectracking.healthit.gov/support/browse/LANTERN-251)
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code accomplishes the tasks purpose.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge
is complete.

Create a new branch `<branch name>_gomodupdate`. Run `make update_mods` to update the `go.mod`
and `go.sum` files to point to master. Create a PR. Require one sanity check reviewer.
